### ISSUE TITLE
comparison of different types with ==, compiler says it will always evaluate to false

### ISSUE THREAD

**Issue by lhk** [microsoft/TypeScript#37381]
**TypeScript Version:**  3.8.3

**Search Terms:**
 - comparison
 - comparisons
 - ==

**Code**

```ts
let a = "1";
let b = 1;

console.log(a == b);
```

**Expected behavior:**
Should complain about incompatible types, maybe with a warning that the `==` operator will lead to an implicit type conversion.

**Actual behavior:**
It does complain, but the error message is wrong.

`error TS2367: This condition will always return 'false' since the types 'string' and 'number' have no overlap.`

This is wrong. If the code is executed, the comparison will evaluate to true.

**Related Issues:**
#30655. In that issue the expected behaviour was that loose comparisons of strings and numbers are considered valid.  The issue was closed with a comment that an error message is intended here. I do agree with an error being the intended output. It's simply that the error message is wrong.

#29155 has exactly this problem as title but seems unrelated.

**Playground Link:** [Provided](https://www.typescriptlang.org/play/index.html?ssl=4&ssc=21&pln=1&pc=1#code/DYUwLgBAhhC8ECICMCDcAoUkBGcJI3QGMB7AOwGcTQA6YEgcwAoZZ5sBKVIA)

**Comment by WhileTrueEndWhile** [comment]
This should be correct. If you do not use `any` compiled code will behave right, if it compiles :-). You should not be able to compare different types. So, implicit conversions should not be valid typescript.

**Comment by lhk** [comment]
Yes, implicit conversions should not be valid typescript. I do agree.

But the error message is wrong. The condition will not always return false, just look at the playground example (where it returns true).

**Comment by nmain** [comment]
Closely related to https://github.com/microsoft/TypeScript/issues/37251

**Comment by RyanCavanaugh** [comment]
Exact duplicate of #26592
